### PR TITLE
[FIX] analytic, sale: timesheet access rights

### PR DESCRIPTION
**Pull request by simongoffin**
A user with employee rights should be able to create a timesheet line.

opw:658083

### CODE REVIEW

**Comment by simongoffin** [comment]
@tac-odoo

**Comment by beledouxdenis** [comment]
Not ok to me.

Employees should be able to create time-sheet lines, indeed.

Nevertheless, they should not be allowed to create analytic accounts (`model_account_analytic_account`).

Regarding `sale.order` & `sale.order.line`, same answers: Only Sales > See Own Leads must be able to create those (and for this specific group, only if the `user_id` of these models are themselves)

I am not sure to follow: Does the creation of a time-sheet line lead to the creation of a `sale order` ? In which conditions ? If this is indeed the case, on purpose, in some conditions, then the sale order should probably be created as `SUPERUSER_ID`. The same can be applied for the `account.analytic.account`

**Comment by simongoffin** [comment]
@nim-odoo

File: addons/sale/sale_analytic.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def write(self, values):
 
         lines = super(AccountAnalyticLine, self).write(values)
         for line in self:
-            res = line._get_sale_order_line(vals=values)
+            res = line.sudo()._get_sale_order_line(vals=values)
             super(AccountAnalyticLine, line).write(res)
 
         self.mapped('so_line')._compute_analytic()
```
Comment (nim-odoo) [inline addons/sale/sale_analytic.py]: missing sudo

**Comment by nim-odoo** [comment]
@beledouxdenis : regarding Sale, registering a timesheet might indeed trigger the creation/modification of SO lines. When an analytic account is linked to a SO, any timesheet line registered will be counted in the 'delivered quantity' of a timesheet product in the SO.

**Comment by beledouxdenis** [comment]
should be done as sudo then :).

**Pull request merged**